src/Bicep.Core/Semantics/Namespaces/DefaultNamespaceProvider.cs

```diff
@@ -14,60 +16,60 @@ namespace Bicep.Core.Semantics.Namespaces;
 public class DefaultNamespaceProvider : INamespaceProvider
```

Comment:
It may be worth segregating the `DefaultNamespaceProvider` from a `DynamicTypeLoadingNamespaceProvider` (better name TBD). `DefaultNamespaceProvider` looks like it only needs an `IResourceTypeProviderFactory` if dynamic loading is enabled (`GetBuiltInAzResourceTypesProvider()` could be a static method on the interface) and could be used a singleton unless dynamic type loading is explicitly enabled.

Reply:
Its an interesting idea, can I do it in a separate PR?

Reply:
@jeskew are you thinking about this as a way of de-risking the change, or thinking about code cleanliness (or something else)?

Reply:
I think that would be better as a separate PR.

Reply:
@anthony-c-martin I was thinking more about efficiency and code cleanliness. Not something that needs to be tackled in this PR